apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,385 @@
+                .border(width = 1.dp, color = if (pref.state.value) MaterialTheme.colors.primary else Color.Unspecified, shape = RoundedCornerShape(50))
+                .padding(3.dp)
+                .size(sp16)
+        )
+    }
+}
+
+private var lazyListState = 0 to 0
+
+@Composable
+private fun ContactsList(
+    contactActions: @Composable () -> Unit,
+    contactTypes: List<ContactType>,
+    contactListTitle: String ? = null
+) {
+    val listState = rememberLazyListState(lazyListState.first, lazyListState.second)
+    val searchText = rememberSaveable(stateSaver = TextFieldValue.Saver) { mutableStateOf(
+        TextFieldValue("")
+    ) }
+
+    var searchFocused by remember { mutableStateOf(false) }
+
+    DisposableEffect(Unit) {
+        onDispose {
+            lazyListState =
+                listState.firstVisibleItemIndex to listState.firstVisibleItemScrollOffset
+        }
+    }
+    val showUnreadAndFavorites =
+        remember { ChatController.appPrefs.showUnreadAndFavorites.state }.value
+
+    val allChats by remember(chatModel.chats, contactTypes) {
+        derivedStateOf { contactChats(chatModel.chats, contactTypes) }
+    }
+
+    val filteredContactChats = filteredContactChats(
+        showUnreadAndFavorites = showUnreadAndFavorites,
+        searchText = searchText.value.text,
+        contactChats = allChats
+    )
+
+    LazyColumn(
+        Modifier.fillMaxWidth(),
+        listState
+    ) {
+        item {
+            SectionView {
+                Divider()
+                ContactsSearchBar(
+                    listState = listState,
+                    searchText = searchText,
+                    focused = searchFocused,
+                    onFocusChanged = {
+                        searchFocused = it
+                    }
+                )
+                Divider()
+            }
+
+            if (!searchFocused) {
+                contactActions()
+            }
+
+            Spacer(Modifier.height(DEFAULT_PADDING))
+
+            if (contactListTitle != null && filteredContactChats.isNotEmpty()) {
+                Text(
+                    contactListTitle, color = MaterialTheme.colors.secondary, style = MaterialTheme.typography.body2,
+                    modifier = Modifier.padding(start = DEFAULT_PADDING, bottom = 5.dp), fontSize = 12.sp
+                )
+            }
+        }
+        itemsIndexed(filteredContactChats) { index, chat ->
+            val nextChatSelected = remember(chat.id, filteredContactChats) {
+                derivedStateOf {
+                    chatModel.chatId.value != null && filteredContactChats.getOrNull(index + 1)?.id == chatModel.chatId.value
+                }
+            }
+            SectionItemView(padding = PaddingValues(horizontal = DEFAULT_PADDING)) {
+                ContactListNavLinkView(chat, nextChatSelected)
+            }
```

Comment:
tap area shouldn't be padded from borders of screen, it may be this or some other padding (i.e. when you tap/hold contact nav link view gray tap background should spread up to border)